### DISCUSSION TITLE
Is it safe to override a scala-xml 1.x dependency with 2.x?

### DISCUSSION THREAD

**Discussion by SethTisue** [scala/scala-xml#605] [Q&A]
This has become something of an FAQ:

> is v2 binary compatible with v1?

> scala-xml is a transitive dependency of a number of my library dependencies, and I'm finding it harder and harder to avoid version conflicts. many newer versions of the libraries I use have migrated to v2, but a number are still on v1

**Comment by SethTisue** [comment]
2.x isn't 100% binary compatible with 1.x (as the major version bump implies), because a bunch of old stuff was removed, as detailed in the various post-1.3.0 release notes at https://github.com/scala/scala-xml/releases/ 

If your particular application isn't using any of that removed stuff, you should be okay. But note that it's hard to be sure if anything in your entire dependency tree was using any of the removed stuff, even if you weren't using that stuff directly yourself.

In general, the removed stuff was pretty obscure, so it's entirely plausible that you can force the 2.0 upgrade and everything will be fine. Whether you're willing to risk doing that may depend on how high the stakes are for you and how much confidence your own test coverage gives you.

Y'all are hereby encouraged to use this discussion thread as a place to share experiences, concerns, workarounds, etc etc.

**Reply by rossabaker** [reply]
In http4s, we're maintaining a separate scala-xml-1 artifact because other dependencies (e.g., Twirl) haven't upgraded.

**Comment by SethTisue** [comment]
@rossabaker Maven Central coordinates?

**Reply by rossabaker** [reply]
```scala
"org.http4s" %% "http4s-scala-xml-1" % "0.23.0"
```

**Comment by mrdziuban** [comment]
All twirl tests pass on scala-xml 2.1.0 -- https://github.com/playframework/twirl/pull/525. YMMV and of course I understand it's a different story for library maintainers, but this was enough to convince me that forcing the upgrade in my application was okay.

**Comment by SethTisue** [comment]
@ckipp01 has made a discussion thread about this specifically in the context of sbt plugins: https://github.com/sbt/sbt/discussions/7075

**Comment by SethTisue** [comment]
If you're affected by the removal of `scala.xml.pull` specifically, see https://github.com/scala/scala-xml/discussions/638
